runtime/domain.c

```diff
@@ -821,15 +885,32 @@ CAMLexport void (*caml_leave_blocking_section_hook)(void) =
    caml_leave_blocking_section_default;
 
 CAMLexport void caml_leave_blocking_section() {
-  caml_plat_lock(&domain_self->roots_lock);
+  dom_internal* self = domain_self;
+
+  atomic_store_rel(&self->backup_thread_msg, MSG_ENTERING_OCAML);
+
```

Comment:
Is it worth having an assert that if `!caml_domain_alone()` then the backup thread is running?

Reply:
done.